store/sqlstore/supplier.go

```diff
@@ -1004,6 +1004,23 @@ func (ss *SqlSupplier) GetAllConns() []*gorp.DbMap {
 	return all
 }
 
+// RecycleDBConnections closes active connections by setting the max conn lifetime
+// to d, and then resets them back to their original duration.
+func (ss *SqlSupplier) RecycleDBConnections(d time.Duration) {
+	// Get old time.
+	originalDuration := time.Duration(*ss.settings.ConnMaxLifetimeMilliseconds) * time.Millisecond
+	// Set the max lifetimes for all connections.
+	for _, conn := range ss.GetAllConns() {
+		conn.Db.SetConnMaxLifetime(d)
+	}
+	// Wait for that period with an additional 2 seconds of scheduling delay.
+	time.Sleep(d + 2*time.Second)
```

Comment:
Would it makes sense to hard-code the various durations entirely within this method? Not sure the caller has the context necessary to provide a meaningful value for `d`.

(If this was just added to enable a shorter unit test, then perhaps we can export the recommended duration value from this package for use by the `app` layer?)

Reply:
> (If this was just added to enable a shorter unit test, then perhaps we can export the recommended duration value from this package for use by the app layer?)

mmm .. I would prefer not to export a package level constant only to be overriden by the tests.

To me, it gives a certain amount of control to the caller. The context is the amount of grace period that can be allowed for connections to be reaped. This wasn't the case originally, and we were sleeping just for a random period of time. In future, we can even add a query param to the API call and pass that here.

That was my reasoning to pass the value from upper layer. I will add a comment to explain this.

 It is an important knob because it gives us control over how fast/slow to reap connections. This all depends on what the customer wants to do / the kind of workload.